sdk/identity/identity/migration-v1-v2.md

```diff
@@ -0,0 +1,309 @@
+
+A sample using the `DeviceCodeCredential` follows. Keep in mind that this example uses the `@azure/identity-cache-persistence` package, which we will discuss in more detail in the [plugins](#plugins) section:
+
+```ts
+import { DeviceCodeCredential, serializeAuthenticationRecord,  deserializeAuthenticationRecord, useIdentityPlugin } from "@azure/identity";
+import { cachePersistencePlugin } from "@azure/identity-cache-persistence";
+
+useIdentityPlugin(cachePersistencePlugin);
+
+// An in-memory example of how the account information can be saved in a serialized way.
+let accountCache: string;
+
+async function main() {
+  // First we need to configure our shared persistence layer
+  const tokenCachePersistenceOptions: TokenCachePersistenceOptions = {
+    enabled: true,
+    name: "my-shared-persistence"
+  };
+
+ const firstCredential = new DeviceCodeCredential({
+    // To be able to re-use the account, the Token Cache must also have been provided.
+    tokenCachePersistenceOptions
+  });
+
+  const account = await firstCredential.authenticate(`https://servicebus.azure.net/.default`);
+
+  accountCache = serializeAuthenticationRecord(account);
+  
+  // The `account` variable could be used directly, but to showcase the serialization, we will deserialize the `accountCache` string variable.
+
+  const secondCredential = new DeviceCodeCredential({
+    // authenticationRecord: account // could also be used
+    authenticationRecord: deserializeAuthenticationRecord(accountCache),
+    // To be able to re-use the account, the Token Cache must also have been provided.
+    tokenCachePersistenceOptions
+  });
+
+  // This getToken method will use the already retrieved credential. It will retrieve it from the cache, and it won't prompt users to authenticate using the typical device code interactive flow.
+  const accessToken = await secondCredential.getToken(`https://servicebus.azure.net/.default`)
+  console.log({ accessToken });
+}
+
+main();
+```
+
+### Changes to ClientSecretCredential and ClientCertificateCredential
+
+For our version two of the `@azure/identity` package, we've added regional STS support to the `ClientSecretCredential` and the `ClientCertificateCredential`. First, we've added a new enum type called `RegionalAuthority`, which contains values for all of the possible regions supported by Azure. Then, we've added a new property called `regionalAuthority` to the `ClientSecretCredentialOptions` and the `ClientCertificateCredentialOptions`, however the Azure region can also be specified through the `AZURE_REGIONAL_AUTHORITY_NAME` environment variable. If instead of a region, `AutoDiscoverRegion` is specified as the value for `regionalAuthority`, MSAL will be used to attempt to discover the region.
```

Comment:
```suggestion
For version 2 of the `@azure/identity` package, we've added regional STS support to the `ClientSecretCredential` and the `ClientCertificateCredential`. First, we've added a new enum type called `RegionalAuthority`, which contains values for all of the possible regions supported by Azure. Then, we've added a new property called `regionalAuthority` to the `ClientSecretCredentialOptions` and the `ClientCertificateCredentialOptions`. However, the Azure region can also be specified through the `AZURE_REGIONAL_AUTHORITY_NAME` environment variable. If instead of a region, `AutoDiscoverRegion` is specified as the value for `regionalAuthority`, MSAL will be used to attempt to discover the region.
```